zuul/connection/github.py

```diff
@@ -430,15 +460,15 @@ def maintainCache(self, relevant):
 
     def getGitUrl(self, project):
         if self.git_ssh_key:
-            return 'ssh://git@%s/%s.git' % (self.git_host, project)
+            return 'ssh://git@%s/%s.git' % (self.git_host, project.name)
 
-        installation_key = self._get_installation_key()
-        if installation_key:
+        if self.integration_id:
+            installation_key = self._get_installation_key(project.name)
             return 'https://x-access-token:%s@%s/%s' % (installation_key,
                                                         self.git_host,
-                                                        project)
+                                                        project.name)
 
-        return 'https://%s/%s' % (self.git_host, project)
+        return 'https://%s/%s' % (self.git_host, project.name)
```

Comment:
Might need to make these `project.get('name')`, since in the test data this is a plain dict, and not a github3.py object.